File: fileOrganiser.js
let fs = require('fs')
let path = require('path')
let a = process.argv.slice(2)

let Types = {
    Music : ['mp3'],
    Video : ['mp4'],
    Document : ['pdf', 'doc', 'docx', 'xlsx', 'ppt', 'pptx', 'txt'],
    Image : ['jpg', 'jpeg', 'jfif', 'png', 'pjpeg' , 'pjp', 'svg', 'webp'],
    Archive : ['zip', 'tar', 'rar']
}

organiseFn(a[0])

function organiseFn(srcPath) {
    if (srcPath == undefined) {
        console.log('Please Enter path')
    } else {
        let doesExists = fs.existsSync(srcPath)
        if (doesExists) {
            let dest = path.join(srcPath, 'OrganisedFiles')
            if (fs.existsSync(dest) == false) {
                fs.mkdirSync(dest)
            }
            organiseHelper(srcPath, dest)
        } else {
            console.log('Path does not exit')
        }
    }
}

function organiseHelper(src, dest) {
    let names = fs.readdirSync(src)
    for (let i = 0; i < names.length; i++) {
        let filePath = path.join(src, names[i])
        let isFile = fs.lstatSync(filePath).isFile()
        if (isFile) {
            let fileType = getFileType(filePath)
            console.log(names[i], ' belongs to --> ', fileType)
            sendFile(filePath, dest, fileType)
        }
    }
}

function sendFile(filePath, dest, fileType) {
    let destPath = path.join(dest, fileType)
    if (fs.existsSync(destPath) == false) {
        fs.mkdirSync(destPath)
    }
    let filename = path.basename(filePath)
    let destfilepath = path.join(destPath, filename)
    fs.copyFileSync(filePath, destfilepath)
    fs.unlinkSync(filePath)
    console.log(filename, ' Copied to ',  fileType)
}

function getFileType(filePath) {
    let ext = path.extname(filePath).slice(1)
    for (let type in Types) {
        let arr = Types[type]
        for (let i = 0; i < arr.length; i++) {
            if (arr[i] == ext)
                return type
        }
    }
    return "Others"
}

File: FileOrganiser.py
import os

print("Enter path")
path = input()

Types = {
    "Music": ['mp3'],
    "Documents": ['pdf', 'doc', 'docx', 'xlsx', 'txt', 'ppt', 'pptx'],
    "Image" : ['jpg', 'jpeg', 'jfif', 'png', 'pjpeg' , 'pjp', 'svg', 'webp'],
    "Archive" : ['zip', 'rar', 'tar'],
    "video" : ['mp4']
}

def getType(filePath):
    _, ext = os.path.splitext(filePath)
    ext = ext[1:]
    for type in Types:
        for i in Types[type]:
            if i == ext:
                return type
    return "Other"

def transferFile(filePath, filename, dest, fileType):
    dest = os.path.join(dest, fileType)
    if os.path.exists(dest) == False:
        os.mkdir(dest)
    dest = os.path.join(dest, filename)
    os.rename(filePath, dest)



def organiseHelper(src, dest):
    filenames = os.listdir(src)
    for i in filenames:
        if i != "Organised":
            filePath = os.path.join(src, i)
            fileType = getType(filePath)
            print(i, "Belongs to -->", fileType)
            transferFile(filePath, i, dest, fileType)
            print(i, "moved to -->", fileType)


def organise(path):
    if path == None:
        print("Enter path")
    else:
        if os.path.exists(path):
            dest = os.path.join(path, "Organised")
            if os.path.exists(dest) == False:
                os.mkdir(dest)
            organiseHelper(path, dest)
        else:
            print("Please Enter Correct path")
            
organise(path)

